scan/lib/scan/detect_values.rb

```diff
@@ -24,9 +24,22 @@ def self.set_additional_default_values
       default_device_tvos if Scan.project.tvos?
       detect_destination
 
+      default_derived_data
+
       return config
     end
 
+    def self.default_derived_data
+      return unless Scan.config[:derived_data_path].to_s.empty?
+      default_path = Scan.project.build_settings(key: "BUILT_PRODUCTS_DIR")
+      # => /Users/.../Library/Developer/Xcode/DerivedData/app-bqrfaojicpsqnoglloisfftjhksc/Build/Products/Release-iphoneos
+      # We need to go one folder up, since tests don't run inside 'Release-iphoneos'
+      # but in 'Debug-iphonesimulator'
+      default_path = File.join(File.expand_path("..", default_path), "Debug-iphonesimulator")
```

Comment:
I don't know what's the status on https://github.com/fastlane/fastlane/pull/5159 in general, and if we want to go finish things up and merge? 
It seems like the best approach here is to check if the current run is for the simulator or for the device and then use `Debug-iphonesimulator` or `Debug-iphoneos` (or similar)

However the derived data path is set to this, so even for the real device it will just work, it just stores the data in a separate directory. 

I wonder if instead it would make sense to just use a fastlane derived data path. I'll give it some more thought tomorrow.